CONTRIBUTING.md

```diff
@@ -85,39 +103,53 @@ The repo includes code comment based API docs, which can be generated with
+test/certs/         # certs for communicating with java-collector
+test/docker/        # docker files for test environments
+test/java-collector # local collector usable for testing
+test/probes/        # tests for probes
+test/<other-dir>/   # subgroup of tests
+test/utility/       # utility for finding problem with a hanging test
+```
+
+`lib/get-unified-config.js` handles configuration, pulling from the config
+file and the environment. `lib/probe-defaults.js` exports the defaults for
+probes. `lib/index.js` is the startup code.
+
 Individual module instrumentation can be found in `lib/probes/${module}.js`,
-while the corresponding tests can be found at `test/probes/${module}.test.js`
-and benchmarks can be found at `test/probes/${module}.bench.js`.
+while the corresponding tests can be found at `test/probes/${module}.test.js`,
+and supporting components, if present, are in `test/probes/${module}/`.
 
-The default config values are all described in `lib/defaults.js`, and get
-applied to the core module that exposes the custom API in `lib/index.js`.
-The lower-level `Span` and `Event` types are described in `lib/span.js`
-and `lib/event.js`.
+## 2 minute overview
 
-The patching mechanism works by intercepting `require(...)` calls in
-`lib/require-patch.js`. The require patch interface includes a `register(...)`
-function, which could be useful for testing patches outside of the appoptics
-module before merging into the core project.
+`/lib/index.js` runs when the user's code requires `appoptics-apm`. It uses
+`lib/get-unified-config.js` to determine the configuration and sets up the
+agent. Eventually it reads the files in `lib/probes` and enables patching.
+
+The patching mechanism works by intercepting `require(...)` calls. It is in
+`lib/require-patch.js`.
 
 Tests live in the `test` directory, with a `test/probes` subdirectory for tests
 specific to a given instrumented module. The file in `test` follow a naming
 scheme of `${name}.test.js` for files intended to be run by the test runner,
-`${name}.bench.js` for files intended to be run the benchmark runner, and files
-with the normal scheme of `${name}.js` are just meant to be used by other files
+files with the normal scheme of `${name}.js` are just meant to be used by other files
 in the test directory.
 
 ## Process
 
 ### Baseline
 
-The code is written with JavaScript features supported by node v6 and above. Until
-the minimum supported version is 8 this means, for example, that `async/await` can't
-be used because it generates JavaScript syntax errors.
+The code is written with JavaScript features supported by node v8 and above. At this
+time the lowest version of node officially supported in v10, so additional features
+maybe be used going forward.
 
 ### Testing
 
-This is an abbreviated version of the testing section above so that it falls in the
-develop, test, release sequence.
+mocha is not a devDependency because mocha is a installed globally in our
+development environment.
```

Comment:
Should `mocha` be mentioned in the dev environment section then?

Reply:
yes

Reply:
added